test/schema_test.cc

```diff
@@ -490,3 +503,474 @@ TEST(SchemaTest, NestedDuplicateFieldIdError) {
+    auto element_field = iceberg::SchemaField{53, "element", struct_type, false};
+    auto list_type = std::make_shared<iceberg::ListType>(element_field);
+    auto list_field = iceberg::SchemaField{54, "list_field", list_type, true};
+    return MakeSchema(list_field);
+  }
+
+  static std::shared_ptr<iceberg::Schema> ListOfMapSchema(int32_t schema_id = 800) {
+    auto map_value_struct = MakeStructType(TestFields::Name(), TestFields::Age());
+    auto map_value_field = iceberg::SchemaField{64, "value", map_value_struct, false};
+    auto map_type =
+        std::make_shared<iceberg::MapType>(TestFields::Key(), map_value_field);
+    auto list_element = iceberg::SchemaField{65, "element", map_type, false};
+    auto list_type = std::make_shared<iceberg::ListType>(list_element);
+    auto list_field = iceberg::SchemaField{66, "list_field", list_type, true};
+    return MakeSchema(list_field);
+  }
+
+  static std::shared_ptr<iceberg::Schema> ComplexMapSchema(int32_t schema_id = 900) {
+    auto key_id_field = iceberg::SchemaField{71, "id", iceberg::int32(), false};
+    auto key_name_field = iceberg::SchemaField{72, "name", iceberg::string(), false};
+    auto key_struct = MakeStructType(key_id_field, key_name_field);
+    auto key_field = iceberg::SchemaField{73, "key", key_struct, false};
+
+    auto value_id_field = iceberg::SchemaField{74, "id", iceberg::int32(), false};
+    auto value_name_field = iceberg::SchemaField{75, "name", iceberg::string(), false};
+    auto value_struct = MakeStructType(value_id_field, value_name_field);
+    auto value_field = iceberg::SchemaField{76, "value", value_struct, false};
+
+    auto map_type = std::make_shared<iceberg::MapType>(key_field, value_field);
+    auto map_field = iceberg::SchemaField{77, "map_field", map_type, true};
+    return MakeSchema(map_field);
+  }
+};
+
+}  // namespace
+
+struct SelectTestParam {
+  std::string test_name;
+  std::function<std::shared_ptr<iceberg::Schema>()> create_schema;
+  std::vector<std::string> select_fields;
+  std::function<std::shared_ptr<iceberg::Schema>()> expected_schema;
+  bool should_succeed;
+  std::string expected_error_message;
+  bool case_sensitive = true;
+};
+
+class SelectParamTest : public ::testing::TestWithParam<SelectTestParam> {};
+
+void CompareSchema(std::shared_ptr<iceberg::Schema> actual_schema,
+                   std::shared_ptr<iceberg::Schema> expected_schema) {
+  ASSERT_EQ(actual_schema->fields().size(), expected_schema->fields().size());
+  for (int i = 0; i < actual_schema->fields().size(); i++) {
+    ASSERT_EQ(actual_schema->fields()[i], expected_schema->fields()[i]);
+  }
+}
+
+TEST_P(SelectParamTest, SelectFields) {
+  const auto& param = GetParam();
+  auto input_schema = param.create_schema();
+  auto result = input_schema->Select(param.select_fields, param.case_sensitive);
+
+  if (param.should_succeed) {
+    ASSERT_TRUE(result.has_value());
+    CompareSchema(std::move(result.value()), param.expected_schema());
```

Comment:
We don't need the `CompareSchema`. Schema has implemented `operator==`.
```suggestion
    ASSERT_EQ(*result.value(), *param.expected_schema());
```